linera-core/src/chain_worker/state/temporary_changes.rs

```diff
@@ -209,30 +208,19 @@ where
         self.0.chain.remove_bundles_from_inboxes(block).await?;
         // Verify that all required bytecode hashed certificate values and blobs are available, and no
         // unrelated ones provided.
+        let published_blob_ids = block.published_blob_ids();
         self.0
-            .check_no_missing_blobs(block.published_blob_ids(), blobs)
+            .check_no_missing_blobs(published_blob_ids.clone(), blobs)
```

Comment:
I know this isn't introduced in this PR but:
- `published_blob_ids` are extracted from the block
- `blobs` are also part of the `BlockProposal`

are they not related? i.e. aren't `blobs` the ones that are referenced in the `block.published_blob_ids()`?

Reply:
Yes, that's why we're checking (in `get_missing_blobs`) that all `blobs` are actually among the ones published in this block, and conversely, that all blobs published in this block are either already known to us or among the provided `blobs`.

Reply:
I think it's wrong. The `Block` should carry just the IDs and, as part of fetching dependencies, we _request_ missing blocks. Then, here, we don't need this check at all.

Reply:
Yes, that would fit Linera better since there's no intra-validator gossiping.